public/js/controllers/MainController.js

```diff
@@ -1,64 +1,88 @@
- /* jshint esversion: 6 */ 
+ /* jshint esversion: 6 */
+ /* eslint no-param-reassign: ["error", { "props": false }]*/
+
  app.controller('MainController', ['$scope', '$http', '$localStorage',
-   function($scope, $http, $localStorage) {
+   ($scope, $http, $localStorage) => {
+     $scope.noFiles = true;
      $scope.fileReady = false;
-     $scope.afile = {};
-     $scope.files = [];
-
-     // Array that holds the checked files
-     $scope.checked = [];
+     $scope.files = []; // Contains all uploaded file names
+     $scope.checked = []; // Contains all the checked files names
 
-     // Get index fuction
-     // @params file specified
-     // returns the index of the file
-     $scope.fetchIndex = function(file) {
+     /**
+      * createIndex
+      *
+      * sends a http.post to server with the file.
+      * @params {string}
```

Comment:
### Nitpick

There should have been a line space before the method description.